[Feat]: Added Pre-loader and custom cursor

shreyash3087 opened the pull request:
**Description:**

This PR adds a pre-loader and a custom cursor on the **prithwe** website,the loader and the cursor creates a much better user experience.

**Issue Addressed:**

Closes: #65 

**Changes Made:**

- Added `cursor.js` to handle custom cursor logic.
- Created `Spinner.jsx` component for the pre-loader animation.
- Updated `App.jsx` to include the Spinner component .
- Modified `App.css` to style the pre-loader.
- Updated `index.html` to integrate the custom cursor.

**Additional Context-** 

[X] I am a part of GSSOC Contributors

Thankyou

shreyash3087 commented:
@adarsh-singh01 if there is no issue,kindly merge it

adarsh-singh01 commented:
> @adarsh-singh01 if there is no issue,kindly merge it

Merged!

shreyash3087 commented:
@adarsh-singh01 I noticed that the cursor doesn't work on deployement although it does work locally, this was because i forgot to give the type of cursor.js as module and therefore render isn't able to compile it as a js file.
please fix this error by adding `type=module` in index.html

![Screenshot (187)](https://github.com/adarsh-singh01/PrithWe/assets/72603662/b8eaad05-3c2e-4843-96a6-17349d14803a)

I deployed the project on render myself to test if adding `type=module` works or not,and it does-

![Screenshot (188)](https://github.com/adarsh-singh01/PrithWe/assets/72603662/4e510860-f55b-42b7-a799-77df1de4847d)

It was a mistake from my end and i apologize for that, adding "type=module" might solve this error, if you think the error might be due to any other reasons,please do express your reasonings.

Thankyou

adarsh-singh01 commented:
Okay... Cursor error has been fixed...can you tell me... is that padding that has came on every page been caused by any code of yours?

shreyash3087 commented:
@adarsh-singh01 nope,the padding was already there, none of my code has any effect on the padding of the website, although since you have mentioned it i can raise another issue and fix it as well

adarsh-singh01 commented:
Hey can you disappear the cursor on small screens(tab, mobile) and make it transparent in big screens.

shreyash3087 commented:
@adarsh-singh01 alright,will do it,do you want me to make a new pr for that or what? as this one is already merged?

adarsh-singh01 commented:
Make a new PR...

shreyash3087 commented:
@adarsh-singh01 just started working on this,but i am a but confused clarify this to me please,you want the cursor effect(the circular trail) to be gone for smaller devices? and the actual pointer cursor to diappear for larger devices, right?

adarsh-singh01 commented:
> @adarsh-singh01 just started working on this,but i am a bit confused clarify this to me please,you want the cursor effect(the circular trail) to be gone for smaller devices? and the actual pointer cursor to diappear for larger devices, right?

i do not want the circular cursor in small devices as it is annoying in small devices...and i do not want the arrow cursor in big devices ,and i want circle cursor(along with trail effect) to be transparent in big screen

adarsh-singh01 commented:
> > @adarsh-singh01 just started working on this,but i am a bit confused clarify this to me please,you want the cursor effect(the circular trail) to be gone for smaller devices? and the actual pointer cursor to diappear for larger devices, right?
> 
> i do not want the circular cursor in small devices as it is annoying in small devices...and i do not want the arrow cursor in big devices ,and i want circle cursor(along with trail effect) to be transparent in big screen

still any confusion?...ask again

shreyash3087 commented:
@adarsh-singh01 alright,will do it

shreyash3087 commented:
@adarsh-singh01 Sorry for the delay i got busy with some other stuff-

I decresed the opacity of the circular cursor but its not quite transparent because there are 7 circles overlapping each other in idle case,so even though their opacity is 50% they dont look quite transparent,but other than that i removed the arrow from big screens( for screen sizes of width:`768` and lower ),if you want the width to be lower,can do that.Also I Removed the cursor for smaller screens,
Please check it out-

https://github.com/adarsh-singh01/PrithWe/assets/72603662/3b9aeca5-ed03-4bd5-b656-a318d1746311

If there is something that needs to be modified,please do tell.

P.S- I modified the cursor script so that it triggers on scroll event too,as earlier it just triggers on mouseMovement you can check that in the website.
